scaleway/data_source_rdb_privilege.go

```diff
@@ -20,8 +20,13 @@ func dataSourceScalewayRDBPrivilege() *schema.Resource {
 }
 
 func dataSourceScalewayRDBPrivilegeRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	instanceID, _ := d.Get("instance_id").(string)
+	region, instanceID, err := parseRegionalID(d.Get("instance_id").(string))
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Does this work with
```hcl
data "scaleway_rdb_privilege" "main" {
  region = "fr-par"
  instance_id   = "11111111-1111-111111111111"
  user_name     = "my-user"
  database_name = "my-database"
}
```

Is regional ID required ? It should not be

Reply:
If fixed, data source example in doc should also be fixed.
We should avoid locality IDs in examples
https://github.com/scaleway/terraform-provider-scaleway/pull/1721/files#diff-c620de89f9ac495bb7a44a54b2223dd12308a81aefccf29ed9f1420f12613e1aR17-R21

Reply:
Thanks for the review. Many changes on the last commit in order to handle the migration on schema.